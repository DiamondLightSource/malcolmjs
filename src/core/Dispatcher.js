/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
