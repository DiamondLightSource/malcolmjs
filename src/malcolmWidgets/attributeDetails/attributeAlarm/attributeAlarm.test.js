import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import AttributeAlarm, { AlarmStates } from './attributeAlarm.component';

describe('AttributeAlarm', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('renders no_alarm correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.NO_ALARM} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders minor_alarm correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm
        alarmSeverity={AlarmStates.MINOR_ALARM}
        theme={{ alarmState: { warning: '#fff' } }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders major_alarm correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.MAJOR_ALARM} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders invalid_alarm correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.INVALID_ALARM} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders undefined_alarm correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.UNDEFINED_ALARM} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders pending correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.PENDING} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders dirty correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.DIRTY} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders dirty & error correctly', () => {
    const wrapper = shallow(
      <AttributeAlarm alarmSeverity={AlarmStates.DIRTYANDERROR} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders blank if no alarm severity specified', () => {
    const wrapper = shallow(<AttributeAlarm />);
    expect(wrapper).toMatchSnapshot();
  });
});
