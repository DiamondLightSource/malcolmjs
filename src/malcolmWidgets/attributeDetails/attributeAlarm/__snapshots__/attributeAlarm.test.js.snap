// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeAlarm renders blank if no alarm severity specified 1`] = `
<div
  style={
    Object {
      "width": 24,
    }
  }
/>
`;

exports[`AttributeAlarm renders dirty & error correctly 1`] = `
<div>
  <svg
    height="26"
    version="1.1"
    viewBox="0 0 48 48"
    width="26"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m24 4c-11.04 0-20 8.95-20 20s8.96 20 20 20 20-8.95 20-20-8.96-20-20-20zm7.5527 8c0.34172 0 0.68332 0.13059 0.94336 0.39062l3.1133 3.1133c0.52007 0.52007 0.52007 1.3666 0 1.8867l-2.4395 2.4414-5.002-5.002 2.4414-2.4395c0.26004-0.26004 0.60164-0.39062 0.94336-0.39062zm-4.7969 4.2441 5 5-14.756 14.756h-5v-5l14.756-14.756z"
      fill="#f44336"
    />
  </svg>
</div>
`;

exports[`AttributeAlarm renders dirty correctly 1`] = `
<pure(Edit)
  nativeColor="#7986cb"
/>
`;

exports[`AttributeAlarm renders invalid_alarm correctly 1`] = `
<pure(HighlightOff)
  nativeColor="#ac00ff"
/>
`;

exports[`AttributeAlarm renders major_alarm correctly 1`] = `
<pure(Error)
  nativeColor="#ff0f03"
/>
`;

exports[`AttributeAlarm renders minor_alarm correctly 1`] = `
<pure(Warning)
  nativeColor="#fff"
/>
`;

exports[`AttributeAlarm renders no_alarm correctly 1`] = `
<pure(InfoOutline)
  nativeColor="#7986cb"
/>
`;

exports[`AttributeAlarm renders pending correctly 1`] = `
<WithStyles(CircularProgress)
  size={24}
  style={
    Object {
      "color": "#ff4081",
    }
  }
/>
`;

exports[`AttributeAlarm renders undefined_alarm correctly 1`] = `
<pure(HighlightOff)
  nativeColor="#ac00ff"
/>
`;
